src/torchmetrics/utilities/enums.py

```diff
@@ -76,8 +46,20 @@ class AverageMethod(EnumStr):
     SAMPLES = "samples"
 
 
-class MDMCAverageMethod(EnumStr):
+class MDMCAverageMethod(StrEnum):
     """Enum to represent multi-dim multi-class average method."""
 
     GLOBAL = "global"
     SAMPLEWISE = "samplewise"
+
+
+class ClassificationTask(StrEnum):
+    """Enum to represent the different tasks in classification metrics.
+
+    >>> "binary" in list(ClassificationTask)
+    True
+    """
+
+    BINARY = "binary"
+    MULTICLASS = "multiclass"
+    MULTILABEL = "multilabel"
```

Comment:
We can also potentially get rid off individual `ValueError` raising if we override `from_str` method as I did for `InfoLM` metric, but not sure if desired.

```
@classmethod
    def from_str(cls, value: str) -> Optional["EnumStr"]:
        """
        Raises:
            ValueError:
                If required task is not among the supported options.
        """
        _allowed_tasks = [task.lower() for tak in ClassificationTask._member_names_]

        enum_key = super().from_str(value)
        if enum_key is not None:
            return enum_key
        raise ValueError(f"Expected `task` argument to be one of {_allowed tasks}, but got {value}.")
        ```
  and then calling `task = ClassificationTask.from_str(task)` will raise a `ValueError` automatically.

Reply:
done now :)